src/AzureBlobStorageUpdateSession/UpdateSession/UpdateSessionTable.cs

```diff
@@ -0,0 +1,123 @@
+
+        /// <summary>
+        /// </summary>
+        /// <param name="connectionString"></param>
+        /// <param name="ttl">If an instance update session is older than ttl, it will be ignored. The goal of this parameter 
+        /// is to avoid situations where an instance fails to end its update session and blocks the cluster.</param>
+        public UpdateSessionTable(string connectionString, TimeSpan ttl)
+        {
+            _ttl = ttl;
+            CloudTableClient client = CloudStorageAccount.Parse(connectionString).CreateCloudTableClient();
+            _table = client.GetTableReference(UpdateSessionTableName);
+            _table.CreateIfNotExists();
+        }
+
+        public Task<UpdateSessionStatus> FetchUpdateSessionStatus(string clusterId, string appId)
+        {
+            string partitionQuery = CreatePartitionQuery(clusterId, appId);
+
+            TableQuery<UpdateDomainEntity> query = new TableQuery<UpdateDomainEntity>().Where(partitionQuery);
+            UpdateDomainEntity updateDomainEntity = null;
+            UpdateDomainEntity instanceListModifiedEntity = null;
+            List<UpdateDomainEntity> instancesList = new List<UpdateDomainEntity>();
+            foreach (UpdateDomainEntity entity in _table.ExecuteQuery(query))
+            {
+                if (entity.RowKey == UpdateDomainRowKey)
+                {
+                    updateDomainEntity = entity;
+                }
+                else if (entity.RowKey == ModifiedRowKey)
+                {
+                    instanceListModifiedEntity = entity;
+                }
+                else
+                {
+                    if (entity.Timestamp.Add(_ttl) > DateTimeOffset.Now)
+                    {
+                        instancesList.Add(entity);
+                    }
+                }
+            }
+
+            IEnumerable<UpdateDomainEntity> instancesEntities = instancesList;
+            if (updateDomainEntity != null)
+            {
+                // filter out entities that do not match the active update domain (were leftover for some reason)
+                instancesEntities = instancesEntities.Where(entity => entity.UpdateDomain == updateDomainEntity.UpdateDomain);
+            }
+
+            return Task.FromResult(new UpdateSessionStatus(instancesEntities, instanceListModifiedEntity, updateDomainEntity));
+        }
+
+        public async Task<bool> TryExecuteTransaction(UpdateSessionTransaction transaction)
+        {
+            try
+            {
+                await _table.ExecuteBatchAsync(transaction.BatchOperation);
+
+                return true;
+            }
+            catch (StorageException e)
+            {
+                if (e.RequestInformation.HttpStatusCode != 412 
+                    && e.RequestInformation.HttpStatusCode != 409)
+                {
+                    throw;
+                }
+            }
+            return false;
```

Comment:
returning false can only happen for 412 and 409 so why don't we do that right away to keep the code tighter together?